src/generics.rs

```diff
@@ -772,6 +772,27 @@ pub(crate) mod parsing {
 
             let begin = input.fork();
 
+            if cfg!(feature = "full") && input.peek(Token![use]) {
+                input.parse::<Token![use]>()?;
+                input.parse::<Token![<]>()?;
+                while !input.peek(Token![>]) {
```

Comment:
Opted for a manual loop here since I'm not certain `Punctuated` gives me the right API for this.